src/main/cluster-manager.ts

```diff
@@ -95,6 +95,8 @@ export class ClusterManager extends Singleton {
     const entity = catalogEntityRegistry.items[index] as KubernetesCluster;
 
     this.updateEntityStatus(entity, cluster);
+    
+    entity.metadata.labels = Object.assign({}, cluster.labels, entity.metadata.labels);
```

Comment:
Why not just `entity.metadata.labels.workspace = cluster.workspace`?

Reply:
See above (as answered on slack), `cluster.workspace` should really have been `cluster.workspaceId`.